javascript/ql/src/semmle/javascript/security/dataflow/CodeInjectionCustomizations.qll

```diff
@@ -169,13 +169,19 @@ module CodeInjection {
   }
 
   /**
-   * The first argument to `Module.prototype._compile` from the Node.js built-in module `module`,
-   * considered as a code-injection sink.
+   * The first argument to `Module.prototype._compile`, considered as a code-injection sink.
    */
   class ModuleCompileSink extends Sink {
     ModuleCompileSink() {
+      // `require('module').prototype._compile`
       this =
         API::moduleImport("module").getInstance().getMember("_compile").getACall().getArgument(0)
+      or
+      // `module.constructor.prototype._compile`
+      exists(DataFlow::SourceNode moduleConstructor |
+        moduleConstructor = DataFlow::moduleVarNode(_).getAPropertyRead("constructor") and
+        this = moduleConstructor.getAnInstantiation().getAMethodCall("_compile").getArgument(0)
```

Comment:
It's a bit ugly to do the modelling using API graphs above and using source nodes here, but adding `module.constructor` as an API node would have needed changes to the API-graph library, which I didn't think were worth it. Happy to be convinced otherwise, though!

Reply:
An alternative would be to add a custom entry point for `module.constructor`, which can be outside the API graph library as long as it's imported from `javascript.qll`. But yeah, this seems good enough to me.